Add basic implementation of primitives

gonzaiva opened the pull request:
The basic implementation for Model, Sampler and MAML have been added,
along with two launcher scripts:
- One for training a basic classifier
- One for training the MAML algorithm

Currently there's only one specific sampler for MNIST.
The MAML algorithm is not converging to good results, even if a pre-
trained model is used (see save and restored methods in Model), so
further debugging is required.

Commit:
Add basic implementation of primitives

The basic implementation for Model, Sampler and MAML have been added,
along with two launcher scripts:
- One for training a basic classifier
- One for training the MAML algorithm
Currently there's only one specific sampler for MNIST.
The MAML algorithm is not converging to good results, even if a pre-
trained model is used (see save and restored methods in Model), so
further debugging is required.

utkarshjp7 reviewed (approved):
I will try to debug MAML today.  The basic implementations looks good.

ghost commented:
> I will try to debug MAML today. The basic implementations looks good.

Thanks Utkarsh.
I will merge everything here except MAML, and put MAML in its own branch until it works to only keep functional code in master.

Commit:
Remove all the MAML code

The MAML code will be placed in its own branch and merged once it's
fully working.
